File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.go
func subarraySum(nums []int, k int) int {
    mp := make(map[int]int)
    mp[0] = 1
    count := 0
    prefix := 0

    for _, num := range nums {
        prefix += num
        count += mp[prefix - k]
        mp[prefix]++
    }
    return count
}

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var subarraySum = function(nums, k) {
    let map = new Map()
    map.set(0, 1)
    let count = 0
    let prefix = 0

    for (let num of nums) {
        prefix += num
        if (map.has(prefix - k)) {
            count += map.get(prefix - k)
        }
        map.set(prefix, (map.get(prefix) || 0) + 1)
    }
    return count
};